### DISCUSSION TITLE
SQLAlchemy code from draw.io UML

### DISCUSSION THREAD

**Discussion by mbnoimi** [sqlalchemy/sqlalchemy#8264] [Usage Questions]
[Java Code Generator][1] is a fantastic tool for generating Java code from draw.io UML class diagram. It's written in Python but only generates Java code.

Is there a similar tool for generating SQLAlchemy models from draw.io UML class diagram?

  [1]: https://github.com/JanitSri/JavaCodeGenerator

**Comment by zzzeek** [comment]
there's something **very old** that does this [here](https://pypi.org/project/sqlalchemy_schemadisplay/) which was based on [this wiki entry](https://github.com/sqlalchemy/sqlalchemy/wiki/SchemaDisplay).     it still seems to have activity on [github](https://github.com/fschulze/sqlalchemy_schemadisplay) though so it might still work OK.

**Reply by mbnoimi** [reply]
> there's something **very old** that does this [here](https://pypi.org/project/sqlalchemy_schemadisplay/) which was based on [this wiki entry](https://github.com/sqlalchemy/sqlalchemy/wiki/SchemaDisplay). it still seems to have activity on [github](https://github.com/fschulze/sqlalchemy_schemadisplay) though so it might still work OK.

Thank you. The suggested project convert sqlalchemy into diagram. I look for the opposite.

**Reply by zzzeek** [reply]
ah the opposite.   I doubt you'll find that, sorry
